src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java

```diff
@@ -102,10 +103,24 @@ protected SauceCredentials getCredentials() {
      * @throws JSONException Not json returned properly
      */
     public static LinkedHashMap<String, JobInformation> retrieveJobIdsFromSauce(SauceREST sauceREST, Run build) throws JSONException {
-        SauceCredentials credentials = build.getAction(SauceOnDemandBuildAction.class).getCredentials();
+        SauceCredentials credentials = getBuildAction(build).getCredentials();
         return retrieveJobIdsFromSauce(sauceREST, build, credentials);
     }
 
+    /**
+     * @param build The build in progress
+     * @return the {@link SauceOnDemandBuildAction} instance which has been registered with the build
+     *         Can be null
+     */
+    private static SauceOnDemandBuildAction getBuildAction(Run build) {
+        SauceOnDemandBuildAction buildAction = build.getAction(SauceOnDemandBuildAction.class);
+        if (buildAction == null && build instanceof MavenBuild) {
+            //try the parent
+            buildAction = ((hudson.maven.MavenBuild) build).getParentBuild().getAction(SauceOnDemandBuildAction.class);
+        }
+        return buildAction;
+    }
+
```

Comment:
done :). Tested locally looks good. Will test it again in intrev once we merge this.